### PR TITLE
Username dev

### PR DESCRIPTION
**Pull request by ceccoemi**
# I have made things!

## Checklist

- [x] I'm sure that I did not unrelated changes in this pull request
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the [CHANGELOG.rst](https://github.com/lk-geimfari/mimesis/blob/master/CHANGELOG.rst)

## Related issues

The function `username()` in the `Person` provider now support custom templates. The specifications of the template format are written in the doctstring of the function.

I tried not to break the compatibility with the old function, so that if `template=None` or `template='default'` then the returned username is the same as before.

@lk-geimfari what do you think about replacing the type of the exception? I think that an `IndexError` exception is not suitable for a wrong template format, what about throwing a `ValueError` instead?

### CODE REVIEW

File: tests/test_providers/test_person.py [hunk 1]
```diff
@@ -88,17 +85,22 @@ def test_username(self, _person, template):
             'l_d': r'^[a-z]+_[0-9]+$',
             'ld': r'^[a-z]+[0-9]+$',
             'default': r'^[a-z]+\.[0-9]+',
+            None: r'^[A-Za-z]{2,}[\.\-\_]?[0-9]+$',
```
Comment (lk-geimfari) [inline tests/test_providers/test_person.py]: @ceccoemi Are you sure that tests cover all the templates?

File: mimesis/providers/person.py [hunk 2]
```diff
@@ -184,28 +191,21 @@ def username(self, template: Optional[str] = None) -> str:
         if template == 'default':
             template = DEFAULT_TEMPLATE
 
-        if template not in templates:
+        if re.fullmatch(r'[Ul\.\-\_d]*[Ul]+[Ul\.\-\_d]*', template) is None:
```
Comment (lk-geimfari) [inline mimesis/providers/person.py]: @ceccoemi Can we write it like this?

```python
if not re.fullmatch(r'[Ul\.\-\_d]*[Ul]+[Ul\.\-\_d]*', template):
    ... 

```

File: tests/test_providers/test_person.py (hunk 1 above)
Comment (ceccoemi) [reply tests/test_providers/test_person.py]: Yes I have spent a lot of time in building that regex and I am pretty sure that it covers all the templates that are used when `template=None`.

I want to clarify that the regex do not cover all the possible usernames with all the possible templates, but it refers to `(‘U_d’, ‘U.d’, ‘U-d’, ‘UU-d’, ‘UU.d’, ‘UU_d’, ‘ld’, ‘l-d’, ‘Ud’, ‘l.d’, ‘l_d’)`, that are the templates used when `template=None`.

File: mimesis/providers/person.py (hunk 2 above)
Comment (ceccoemi) [reply mimesis/providers/person.py]: Sure! I will fix this

**Comment by lk-geimfari** [comment]
@ceccoemi You can merge it.

**Comment by lk-geimfari** [comment]
@ceccoemi About exception: Sure, you can replace it with much suitable. Don't forget to update docstrings and tests.

**Comment by ceccoemi** [comment]
@lk-geimfari the pull request is ready. Due to my last commit, you have to approve it again.

**Comment by lk-geimfari** [comment]
@ceccoemi You can merge it! 👍

**Pull request merged**